### PR TITLE
 [cms] Update LineItemPayouts to new request InvoicePayouts

### LINKED ISSUE #1339: [cms] Export to CSV seems to be breaking on certain chars
Appears that payout summaries "export to csv" is breaking when "#" are displayed.  

Everything on the csv looks fine until the first "#" then it appears to just encoding the csv.

### PR DESCRIPTION
**Pull request by alexlyp**
Close #1339 

This PR updates the existing line items payout page to use the 
updated invoice payouts request.  Also fixes an issue with "#"
characters causing exporting to CSV to fail.

### CODE REVIEW

**Comment by fernandoabolafio** [comment]
I am not able to submit invoices because of: `1038 current contractor status does not allow new invoices to be created`. How I can set a contractor type for my user so I am able to bypass that?

Also, just as a reminder, in politeiagui we're missing a few CMS error messages. From code 1031 to 1046 IIRC.

**Comment by alexlyp** [comment]
Yah right, there is currently not a edit/manage user endpoint for GUI.  So you'll have to use cmsww (the new cli)  to login as an admin then use the manageuser command to update that user's contractor type to Direct (which is 1 i think)

**Comment by alexlyp** [comment]
And 10-4 on the errors.  There are a bunch of oustanding errors that need to be added along with all of the functionality.  I am going to begin working on them now.

**Comment by fernandoabolafio** [comment]
I am receiving an  "invalid user edit action" while trying to update the contractor domain and type. This is the verbose response from cmswww:
```
cmswww --host="https://localhost:4443" --skipverify -v manageuser 94f305bb-6fb7-4459-8374-c994b073fead  --domain=1
Request: GET https://localhost:4443/v1/user/me
Response: 200
{
  "isadmin": true,
  "userid": "94f305bb-6fb7-4459-8374-c994b073fead",
  "email": "ma@u.com",
  "username": "mau",
  "publickey": "42d494e2a26fd8bf0b5c9b7af38befee6e1cf81e16558c9238e1670f62ed81cb",
  "paywalladdress": "",
  "paywallamount": 0,
  "paywalltxnotbefore": 0,
  "paywalltxid": "ab0af93d4c0a8fcc6db609a882df2452888d53b360a313c2f8e3e8ae658f2a6d",
  "proposalcredits": 0,
  "lastlogintime": 1568972590,
  "sessionmaxage": 86400
}
Request: GET https://localhost:4443/v1/user/94f305bb-6fb7-4459-8374-c994b073fead
Response: 200
{
  "user": {
    "id": "94f305bb-6fb7-4459-8374-c994b073fead",
    "email": "ma@u.com",
    "username": "mau",
    "isadmin": true,
    "identities": [
      {
        "pubkey": "ed7ed0ffe0a1c0f1fecc73d07ddde470a4f0c28a97ffbbfb4db7ae3f2838c9ab",
        "isactive": false
      },
      {
        "pubkey": "13e6cc97d5ea11d320c03f8e71897cf02eb4ddc55f33dc3ea9cc3306d6ad8b65",
        "isactive": false
      },
      {
        "pubkey": "980dfe02d3784497873c3a7cbd18ff9bb03b69c0b258d19e562a627a182d524a",
        "isactive": false
      },
      {
        "pubkey": "cb67afa970cb091d26ba88718865c4ce93aca8752d187afd5f8fcf6af894291e",
        "isactive": false
      },
      {
        "pubkey": "b6a622a381da4052301205ba62182a22ba3f003b274cc560d57568e03ba2929a",
        "isactive": false
      },
      {
        "pubkey": "003ef17a3790b738f30159f9fc40f3852acaa29cbb4177cb70fa837136fed534",
        "isactive": false
      },
      {
        "pubkey": "2dff8873fd2bc90bf6e4779d5576e441fc977cf7ef5b2224a6606cfbb4816f59",
        "isactive": false
      },
      {
        "pubkey": "42d494e2a26fd8bf0b5c9b7af38befee6e1cf81e16558c9238e1670f62ed81cb",
        "isactive": true
      }
    ],
    "lastlogintime": 1568972590,
    "failedloginattempts": 0,
    "isdeactivated": false,
    "islocked": false,
    "emailnotifications": 0,
    "newuserverificationtoken": null,
    "newuserverificationexpiry": 0,
    "updatekeyverificationtoken": null,
    "updatekeyverificationexpiry": 0,
    "resetpasswordverificationtoken": null,
    "resetpasswordverificationexpiry": 0,
    "domain": 0,
    "githubname": "",
    "matrixname": "",
    "contractortype": 0,
    "contractorname": "",
    "contractorlocation": "",
    "contractorcontact": "",
    "supervisoruserid": ""
  }
}
Your current Contractor Type setting is: "0" Update? (n/no/y/yes) [no]: yes
(1) Direct, (2) Supervisor, (3) Sub contractor:  1
Your current Contractor Type setting is: "1" Keep this? (n/no/y/yes) [yes]: yes

Please carefully review your information and ensure it's correct. If not, press Ctrl + C to exit. Or, press Enter to continue your request.
Request: POST https://localhost:4443/v1/user/manage
{
  "userid": "94f305bb-6fb7-4459-8374-c994b073fead",
  "domain": 1,
  "contractortype": 1
}
400, invalid user edit action
```

Do you know what Im doing wrong here?

**Comment by fernandoabolafio** [comment]
Never mind, I fixed. It was a bad setup on politeiawww.

**Comment by alexlyp** [comment]
Hrm I'm unable to reproduce the issue that you're currently facing.  Are you sure you are currently using a tip version of politeiawww?

**Comment by fernandoabolafio** [comment]
I will try it once more against latest master of politeiawww

**Comment by fernandoabolafio** [comment]
Ok, so I tested it again making sure I had politeiawww on latest master. The result is very strange.

While in my network console I receive this as a response to `v1/admin/invoicepayouts`:
Screen Shot 2019-10-05 at 12 20 57

When I log the response from the javascript code I get this:
Screen Shot 2019-10-05 at 12 21 23


Notice how the line items structure has a different value for the label and its overall shape is different.  (For example, the first one doesn't even have a "token" field).

It's a very strange behavior which might just be something wrong on my end so I am ok moving this in it if you say it's working properly for you but I want to make sure there isn't a  hidden bug we are missing.

File: src/selectors/api.js [hunk 1]
```diff
@@ -835,14 +835,34 @@ export const setStatusInvoicePayloadStatus = compose(
...
+  "invoicePayouts"
 );
+export const invoicePayouts = compose(
+  get("invoices"),
+  invoicePayoutsResponse
+);
+
+// TODO: Use createSelector instead of compose once we refactor the CMS connectors
+// b/c the createSelector doesn't work properly whe used in combination with the selectorMap
 export const lineItemPayouts = compose(
```
Comment (fernandoabolafio) [inline src/selectors/api.js]: @alexlyp  I found out what was happening. This was modifying the invoices object directly which was creating side effects all over the data flow. I modified the selector to use a more functional approach and now it is working fine.

Comment (alexlyp) [reply src/selectors/api.js]: Ah that's great.
 Just resolving the small conflict, then it should be good to merge as well.

**Pull request merged**